apis/nucleus/handler/hypercube-handler.js

```diff
@@ -156,62 +241,119 @@ class HyperCubeHandler extends DataPropertyHandler {
+   * If the dimension has a qLibraryId, it will use the library dimension auto-sort.
+   * Otherwise, it will use the field dimension auto-sort.
+   * @memberof HyperCubeHandler
+   * @example
+   * const sortedDimension = hyperCubeHandler.autoSortDimension(dimension);
+   */
   autoSortDimension(dimension) {
     if (dimension.qLibraryId) {
       return getAutoSortLibraryDimension(this, dimension);
     }
-    return getAutoSortFieldDimension(this, dimension);
+    return getAutoSortDimension(this, dimension);
   }
 
   // ----------------------------------
   // ------------ MEASURES ------------
   // ----------------------------------
 
+  /**
+   * @returns {NxMeasure[]} measures
+   * @description Returns the measures of the hypercube.
+   * @memberof HyperCubeHandler
+   * @example
+   * const measures = hyperCubeHandler.getMeasures();
+   */
   getMeasures() {
     return this.hcProperties ? this.hcProperties.qMeasures : [];
   }
 
+  /**
+   * @returns {NxMeasure[]} alternative measures
+   * @description Returns the alternative measures of the hypercube.
+   * @memberof HyperCubeHandler
+   * @example
+   * const alternativeMeasures = hyperCubeHandler.getAlternativeMeasures();
+   */
   getAlternativeMeasures() {
     return this.hcProperties?.qLayoutExclude?.qHyperCubeDef?.qMeasures ?? [];
   }
 
+  /**
+   * @returns {NxMeasureInfo[]} measure layouts
+   * @description Returns the measure layouts of the hypercube.
+   * @memberof HyperCubeHandler
+   * @example
+   * const measureLayouts = hyperCubeHandler.getMeasureLayouts();
+   */
   getMeasureLayouts() {
     const hc = hcUtils.getHyperCube(this.layout, this.path);
     return hc ? hc.qMeasureInfo : [];
   }
 
+  /**
+   * @param {string} cId
+   * @returns {object} measure layout
+   * @description Returns the measure layout of the hypercube for a given cId.
+   * @memberof HyperCubeHandler
+   * @example
+   * const measureLayout = hyperCubeHandler.getMeasureLayout(cId);
+   */
   getMeasureLayout(cId) {
     return this.getMeasureLayouts().filter((item) => cId === item.cId)[0];
   }
 
+  /**
+   * @param {object} measure
+   * @param {boolean} alternative
+   * @param {number} idx
+   * @returns {object} measure
+   * @description Adds a measure to the hypercube.
+   * If the measure is an alternative, it will be added to the alternative measures.
+   * If the total number of measures exceeds the limit, it will stop adding measures.
+   * @memberof HyperCubeHandler
+   * @example
+   * const measure = hyperCubeHandler.addMeasure(measure, alternative, idx);
+   */
   addMeasure(measure, alternative, idx) {
-    const measures = this.getAlternativeMeasure();
-    const meas = initializeId(measure);
+    const meas = initializeField(measure);
 
-    if (hcUtils.isMeasureAlternative(this, measures, alternative)) {
-      return hcUtils.addAlternativeMeasure(this, meas, idx);
+    if (hcUtils.isMeasureAlternative(this, meas, alternative)) {
```

Comment:
should not the second argument to isMeasureAlternative be the list of current measures?